sentry-android-core/src/test/java/io/sentry/android/core/AppLifecycleIntegrationTest.kt

```diff
@@ -94,4 +96,23 @@ class AppLifecycleIntegrationTest {
 
         verify(fixture.handler).post(any())
     }
+
+    @Test
+    fun `When AppLifecycleIntegration is closed from a background thread, watcher is set to null`() {
+        val sut = fixture.getSut(mockHandler = false)
+        val latch = CountDownLatch(1)
+
+        sut.register(fixture.hub, fixture.options)
+
+        assertNotNull(sut.watcher)
+
+        Thread {
+            sut.close()
+            latch.countDown()
+        }.start()
+
+        latch.await()
+        InstrumentationRegistry.getInstrumentation().waitForIdleSync()
```

Comment:
For the main looper it should be possible to use `shadowOf(Looper.getMainLooper()).idle()` instead of creating a thread manually. It would be easier for readability if it works